node/lib/openshift-origin-node/model/v2_cart_model.rb

```diff
@@ -400,7 +400,11 @@ def post_configure(cartridge_name)
           if cartridge.name == primary_cartridge.name and empty_repository?
             output << "CLIENT_MESSAGE: An empty Git repository has been created for your application.  Use 'git push' to add your code."
           else
-            output << start_cartridge('start', cartridge, user_initiated: true)
+            if @state.value == State::STARTED
+              output << start_cartridge('restart', cartridge, user_initiated: true)
```

Comment:
FYI -- restart may not be equivalent to stop/start. Therefore you are introducing a different semantic to post_configure using restart. The issue with calling stop/start is since post_install has not run yet, stop may fail with an error.